egs/ami/s5b/local/ami_sdm_scoring_data_prep.sh

```diff
@@ -122,7 +122,7 @@ if [ `cat $tmpdir/segments_to_fix | wc -l` -gt 0 ]; then
   while read line; do
      p1=`echo $line | awk -F'>' '{print $1}'`
      p2=`echo $line | awk -F'>' '{print $2}'`
-     sed -ir "s:$p1:$p2:" $tmpdir/segments
+     perl -i -pe "s:$p1:$p2:" $tmpdir/segments
```

Comment:
This and the 4 above are an exact match to existing behavior. I would only change the separator to `:` everywhere, the mismatch kinda sticks out (and the `!` is special, but not in non-interactive shell, but still I'd avoid it). Another thing I'd add is, to make sure only really the whole line is replaced, `"s:^$p1\$:$p2:"`. Note that the `$` EOL match assertion needs to be quoted `\n` because shell; try `echo "\$"`.

If it works, it works, you may stop here. But if you look at it, the whole script block is written kinda weirdly, tho. First, a file of complete replacements for the segment file is written to file `segment_replacements` (in another inline perl progam), in the form
```
old line 1>new line 1
old line 2>new line 2
```
etc. Then each line is read inside this loop, split into `p1`=_old line_ and `p2`=_new line_, and a separate pass over the segments file is done to replace them. It maybe ok if done once.

If you really want to improve it, you can change that above perl program to write another little perl program instead, so the `segment_replacements` would be output in the form
```
s/^old line 1$/new line 1/;
s/^old line 2$/new line 2/;
```
and so on. Mind the `;` at ends. Then this whole while loop can be replaced with
```
perl -i -pf segment_replacements $tmpdir/segments
```
which will do a single pass over the segments file. The `-f` is to read program from file `segment_replacements` instead of `-e` for an inline program.

This is all up to you; just how I would do it. Depends on your desire to spend little time hacking perl scripts : ). Srsly, I have an impression that @danpovey puts certain weight on the code's having a didactic value, so it may make sense in the end.

Reply:
Nice catch! I made the change for a single-pass replacement.

Reply:
@kkm000 revisiting this old comment. The `perl` command was not making the required replacement, but when I changed `s/^old line 1$/new line 1/;` to `s/old line 1/new line 1/;` it worked successfully.

Reply:
@desh2608 I am having trouble understanding what you are saying here, perhaps be more specific.  The dollar sign is supposed to match end-of-line.

Reply:
I found that there was a space at the end. After adding that in the regex, it works fine.